### PR TITLE
Master

### PR DESCRIPTION
**Pull request by Dakantz**
Adding a new emscripten js method to parse depth images

### CODE REVIEW

**Comment by farindk** [comment]
Thanks. One thing, though: your PR contains a lot of code reformatting in unrelated parts. Could you please clean up the PR such that only your new code is contained? Also make sure not to merge "master" into your branch. This will make the history unnecessarily complicated.

**Comment by Dakantz** [comment]
Sorry for the merge and formatting mistake, should be fixed now!

This is one of my first PRs 😅 

I hope it can be merged now!

File: libheif/heif_emscripten.h [hunk 1]
```diff
@@ -259,6 +259,95 @@ static emscripten::val heif_js_decode_image2(struct heif_image_handle* handle,
...
+  struct heif_error err = heif_decode_image(depth_handle, &depth_image, heif_colorspace_monochrome, heif_chroma_monochrome, nullptr);
+  if (err.code)
+  {
+    depth_result.set("err", "could not get image of depth image");
+    depth_result.set("msg", err.message);
+    depth_result.set("id", depth_image_id);
+    return depth_result;
+  }
+
+  depth_result.set("testdata", emscripten::val(emscripten::typed_memory_view(10, "123456890")));
```
Comment (farindk) [inline libheif/heif_emscripten.h:290]: looks like this is a leftover from testing

File: libheif/heif_emscripten.h [hunk 2]
```diff
@@ -259,6 +259,95 @@ static emscripten::val heif_js_decode_image2(struct heif_image_handle* handle,
...
+
+      int stride;
+      const uint8_t *plane = heif_image_get_plane_readonly(depth_image, channel, &stride);
+      if (plane)
+      {
+        depth_result.set("stride", stride);
+        depth_result.set("data", emscripten::val(emscripten::typed_memory_view(stride * height, plane)));
+        depth_result.set("buffersize", stride * height);
+        depth_result.set("channel", channel);
+        depth_result.set("testdata", emscripten::val(emscripten::typed_memory_view(10, "123456890")));
```
Comment (farindk) [inline libheif/heif_emscripten.h:317]: another leftover

**Comment by farindk** [comment]
@EzraH442 Could you please do a review on this PR (only the changes in the file `libheif/heif_emscripten.h`.

File: libheif/heif_emscripten.h [hunk 3]
```diff
@@ -259,6 +259,95 @@ static emscripten::val heif_js_decode_image2(struct heif_image_handle* handle,
...
+{
+  emscripten::val depth_result = emscripten::val::object();
+  struct heif_image_handle *depth_handle;
+  heif_image_handle_get_depth_image_handle(handle, depth_image_id, &depth_handle);
+  if (!depth_handle)
+  {
+    depth_result.set("err", "could not get handle of depth image");
+    depth_result.set("id", depth_image_id);
+    return depth_result;
+  }
```
Comment (EzraH442) [inline libheif/heif_emscripten.h:272]: It's better to return an `heif_error` here since heif_error is registered as an value object [see here](https://github.com/strukturag/libheif/blob/a3f43c1b67fbdd9892cc101d231029d4fde10c19/libheif/api/libheif/heif_emscripten.h#L463C1-L463C55), that way it won't be necessary to manually delete the object on the JS side [see here](https://emscripten.org/docs/porting/connecting_cpp_and_javascript/embind.html?highlight=embind#memory-management)

File: libheif/heif_emscripten.h [hunk 4]
```diff
@@ -259,6 +259,95 @@ static emscripten::val heif_js_decode_image2(struct heif_image_handle* handle,
...
+  depth_result.set("height", height);
+
+  struct heif_image *depth_image;
+  struct heif_error err = heif_decode_image(depth_handle, &depth_image, heif_colorspace_monochrome, heif_chroma_monochrome, nullptr);
+  if (err.code)
+  {
+    depth_result.set("err", "could not get image of depth image");
+    depth_result.set("msg", err.message);
+    depth_result.set("id", depth_image_id);
+    return depth_result;
```
Comment (EzraH442) [inline libheif/heif_emscripten.h:287]: same issue as above

File: libheif/heif_emscripten.h [hunk 5]
```diff
@@ -259,6 +259,95 @@ static emscripten::val heif_js_decode_image2(struct heif_image_handle* handle,
...
+        return depth_result;
+      }
+    }
+  }
+
+  depth_result.set("err", "could get no plane of depth image");
+  depth_result.set("id", depth_image_id);
+  return depth_result;
+}
+static emscripten::val heif_js_get_depth_imgs_decoded(struct heif_image_handle *handle)
```
Comment (EzraH442) [inline libheif/heif_emscripten.h:327]: based on what this function does a better name might be heif_js_image_handle_get_depth_img_handle, since it doesn't actually do any image decoding

File: libheif/heif_emscripten.h [hunk 6]
```diff
@@ -259,6 +259,95 @@ static emscripten::val heif_js_decode_image2(struct heif_image_handle* handle,
...
+        depth_result.set("data", emscripten::val(emscripten::typed_memory_view(stride * height, plane)));
+        depth_result.set("buffersize", stride * height);
+        depth_result.set("channel", channel);
+        depth_result.set("testdata", emscripten::val(emscripten::typed_memory_view(10, "123456890")));
+        return depth_result;
+      }
+    }
+  }
+
+  depth_result.set("err", "could get no plane of depth image");
```
Comment (EzraH442) [inline libheif/heif_emscripten.h:323]: This might be a little bit pedantic but I'd rephrase this error message to "could not get plane of depth image"

**Comment by farindk** [comment]
I think you should revert your push of 1870 commits :-)
